### PR TITLE
Pass no_proxy environment variable

### PR DESCRIPTION
**Pull request by JoshuaWatt**
Updates pyrex to pass no_proxy in addition to http_proxy and https_proxy

### CODE REVIEW

**Review by matthoosier-garmin — COMMENTED** [review]
Tactically speaking, this will accomplish what it says on the tin.

I don't see the harm, but what situation motivates this? It doesn't immediately seem to address the internal coming problem inside Garmin, unless you're planning on just setting "no_proxy=*" while still leaving http_proxy set. That seems a little bit indirect; if this is the case, wouldn't just avoiding setting http_proxy in the first place be better?

**Comment by JoshuaWatt** [comment]
> Tactically speaking, this will accomplish what it says on the tin.
> 
> I don't see the harm, but what situation motivates this? It doesn't immediately seem to address the internal coming problem inside Garmin, unless you're planning on just setting "no_proxy=*" while still leaving http_proxy set. That seems a little bit indirect; if this is the case, wouldn't just avoiding setting http_proxy in the first place be better?

Currently we are spamming the logs by using the proxy to access the internal servers, so this should at least reduce the signal to noise ratio, aside from simply being useful in general

**Comment by matthoosier-garmin** [comment]
Okay. I see that wget pays attention to this. Are there other environment variables that serve the same role for stuff like Git's built-in HTTP fetcher, which we should add in here too?

**Comment by JoshuaWatt** [comment]
> Okay. I see that wget pays attention to this. Are there other environment variables that serve the same role for stuff like Git's built-in HTTP fetcher, which we should add in here too?

According to https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ this should give us the best coverage

**Comment by JoshuaWatt** [comment]
> > Okay. I see that wget pays attention to this. Are there other environment variables that serve the same role for stuff like Git's built-in HTTP fetcher, which we should add in here too?
> 
> According to https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ this should give us the best coverage

TL;DR `no_proxy` is supported by pretty much everything, although the semantics slightly vary

**Comment by matthoosier-garmin** [comment]
Okay. If you already know that does what you want, seems fine to me.

**Review by matthoosier-garmin — APPROVED** [review]
